src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -1221,6 +1280,27 @@ private bool ScanNumericLiteral(ref TokenInfo info)
             return true;
         }
 
+        // TODO: Change to Int64.TryParse when it supports NumberStyles.AllowBinarySpecifier (inline this method into GetValueUInt32/64)
+        private bool TryParseBinaryUInt64(string text, out ulong value)
+        {
+            value = 0;
+            for (int i = 0; i < text.Length; i++)
+            {
+                // if uppermost bit is set, then the next bitshift will overflow
+                if ((value & 0x8000000000000000) != 0)
+                {
+                    return false;
+                }
+                if (!SyntaxFacts.IsBinaryDigit(text[i]))
+                {
+                    return false;
```

Comment:
@gafter It is possible if you just consider it as pure function in isolation.
. Does it manipulate anything external to this function, excluding parameters?  
What possible values can a string have?  Is it restricted to 0 and 1 (widechar variants inc) ? No.